plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx

```diff
@@ -46,6 +202,10 @@ export const Edit = ( {
 		[]
 	);
 
+	if ( shouldHideBlock() ) {
+		return null;
+	}
```

Comment:
Just want to note the current behavior of the Overlay Navigation block inside Product Filters: When the block is hidden in the inserter, if users select the block in the list view, they can still see the inspector controls generated by block supports 
<img width="1281" alt="image" src="https://github.com/user-attachments/assets/30fab3c1-0073-4830-ab03-86e19471e33f">

Reply:
Yeah, I believe merchants can edit these blocks even though they are hidden, because if they select "Mobile" or "Always" the block will appear with their customizations.

Reply:
Yeah, but users can edit only settings generated by Supports API, our custom inspector settings aren't rendered when the block is hidden.

Reply:
I got it. I thought you were saying that it should be hidden. I just fixed it, and now the custom settings are being displayed when the block is selected, even though it is hidden a64d44b5c3ac9ce6b98ab4828aa214a93f11c941